core/src/smartcontracts/isi/permissions/has_token.rs

```diff
@@ -78,8 +90,7 @@ impl<H: HasToken> IsAllowed for HasTokenAsValidator<H> {
             ValidatorVerdict::Allow
         } else {
             ValidatorVerdict::Deny(format!(
-                "Account does not have the needed permission token: {:?}.",
-                permission_token
+                "Account does not have the needed permission token: {permission_token:?}",
```

Comment:
Shouldn't this be `Display`?

Reply:
I've choosed to use `Debug` because someone has already decided that `Display` for `PermissionToken` should not expose parameters.

But I can see, that this decision is pretty confusing, so I'll change `impl Display for PermissionToken` and will use `Display` here

Reply:
> I've choosed to use Debug because someone has already decided that Display for PermissionToken should not expose parameters.

that's actually why using `Display` would be a good idea. We might not want to automatically let everyone know what the parameters are. As an option we might enable showing parameters (e.g. using `Debug`) for `#[cfg(debug_assertions)]`.

Reply:
So let's keep `Display` showing parameters for now? If we want just to show the permission token name we can write `name()` method and print it